aspnetcore/signalr/configuration.md

```diff
@@ -227,34 +240,45 @@ let connection = new signalR.HubConnectionBuilder()
     .build();
 ```
 
+
+In the SignalR Java client, you can configure a bearer token to use for authentication by providing an "access token factory" to the [HttpHubConnectionBuilder](/java/api/com.microsoft.signalr._http_hub_connection_builder?view=aspnet-signalr-java). Use [withAccessTokenFactory](/java/api/com.microsoft.signalr._http_hub_connection_builder.withaccesstokenprovider?view=aspnet-signalr-java#com_microsoft_signalr__http_hub_connection_builder_withAccessTokenProvider_Single_String__) to provide an [RxJava](https://github.com/ReactiveX/RxJava) [Single<String>](http://reactivex.io/documentation/single.html). With a call to [Single.defer](http://reactivex.io/RxJava/javadoc/io/reactivex/Single.html#defer-java.util.concurrent.Callable-), you can write logic to produce access tokens for your client.
+
+```java
+HubConnection hubConnection = HubConnectionBuilder.create("https://example.com/myhub")
+    .withAccessTokenProvider(Single.defer(() -> {
+        // Your logic here.
+        return Single.just("An Access Token");
+    })).build();
+```
+
 ### Configure timeout and keep-alive options
 
 Additional options for configuring timeout and keep-alive behavior are available on the `HubConnection` object itself:
 
-| .NET Option | JavaScript Option | Default Value | Description |
-| ----------- | ----------------- | ------------- | ----------- |
-| `ServerTimeout` | `serverTimeoutInMilliseconds` | 30 seconds (30,000 milliseconds) | Timeout for server activity. If the server hasn't sent a message in this interval, the client considers the server disconnected and triggers the `Closed` event (`onclose` in JavaScript). This value must be large enough for a ping message to be sent from the server **and** received by the client within the timeout interval. The recommended value is a number at least double the server's `KeepAliveInterval` value, to allow time for pings to arrive. |
-| `HandshakeTimeout` | Not configurable | 15 seconds | Timeout for initial server handshake. If the server doesn't send a handshake response in this interval, the client cancels the handshake and triggers the `Closed` event (`onclose` in JavaScript). This is an advanced setting that should only be modified if handshake timeout errors are occurring due to severe network latency. For more detail on the Handshake process, see the [SignalR Hub Protocol Specification](https://github.com/aspnet/SignalR/blob/master/specs/HubProtocol.md). |
+| .NET Option | JavaScript Option | Java Option | Default Value | Description |
```

Comment:
We'll probably want to refactor this table now that it's getting bigger.

Reply:
I agree with Brennan. What do you think about using 3 tabs: .NET, JavaScript, and Java?

Reply:
> 3 tabs: .NET, JavaScript, and Java?

Just want to clarify,  under Configuration in the TOC having separate sections for each client? Or something else?

Reply:
I'm imagining actual tabs (see https://docs.microsoft.com/aspnet/core/tutorials/first-web-api?view=aspnetcore-2.2&tabs=visual-studio#create-a-web-project for an example). The syntax for the tabs would be:

```markdown
[.NET](#tab/dotnet)

.NET-specific table here

[JavaScript](#tab/javascript)

JavaScript-specific table here

[Java](#tab/java)

Java-specific table here

---
```

Reply:
Oh, awesome. Thanks for the clarification. 👍

Reply:
ver ver cool

Reply:
Remove this section now

Reply:
I agree with Brennan's suggestion ☝️ . Also remove the 2 sentences below this table.